app/models/spree/gateway/stripe_sca.rb

```diff
@@ -129,7 +129,19 @@ def fetch_payment_intent(creditcard, gateway_options)
         payment = fetch_payment(creditcard, gateway_options)
         raise Stripe::StripeError, I18n.t(:no_pending_payments) unless payment&.response_code
 
-        Stripe::PaymentIntentValidator.new.call(payment.response_code, stripe_account_id)
+        payment_intent_response = Stripe::PaymentIntentValidator.new.
+          call(payment.response_code, stripe_account_id)
+
+        raise_if_not_in_capture_state(payment_intent_response)
+
+        payment.response_code
+      end
+
+      def raise_if_not_in_capture_state(payment_intent_response)
+        state = payment_intent_response.status
+        return unless state != 'requires_capture'
```

Comment:
```suggestion
        return if state == 'requires_capture'
```

Reply:
I just cut-and-pasted this from one place to another, but yeah that's much better :+1: